### PR TITLE
New record: Tune fused softcap kernels and fuse fp8 quantization in LM head (-3.1s)

### PR DESCRIPTION
**Pull request by andrewbriand**
Tune the fused softcap kernels by setting num_warps = 2 and fuse the fp8 quantization of the gradient into the bwd kernel. Improves training time on single H100 from 754s to 728s. I also tried to remove logit softcapping, since the sigmoid is pretty expensive here, but that made training diverge. I'm not sure if that's due to a bug in my implementation or something else.

### CODE REVIEW

**Comment by varunneal** [comment]
> I also tried to remove logit softcapping, since the sigmoid is pretty expensive here, but that made training diverge. 

probably has to do with the fp8 scales, probably w_s if I were to guess

Though i doubt we can get rid of logit softcapping, its pretty good

**Comment by andrewbriand** [comment]
> probably has to do with the fp8 scales, probably w_s if I were to guess

> Though i doubt we can get rid of logit softcapping, its pretty good

You're right on both, with bf16 LM head the training doesn't diverge, but even with 100 extra steps, val loss doesn't get below 3.28.

**Comment by ClassicLarry** [comment]
Feel free to run this whenever against the main train.py. If this isn't touching the logic, don't need to show p value <0.01. I think should remove the bool USE_SOFTCAPPING since its always true.

**Comment by andrewbriand** [comment]
@ClassicLarry I've retimed the current main and this PR:
Main: [96037, 96053, 96101] for an average of 96076 ms
This PR: [92960, 92945, 93052] for an average of 92982 ms

**Comment by ClassicLarry** [comment]
I'm updating the torch version to 2.10 and off of the nightlies. On the new torch, getting 95.7 before and 92.4 after, for -3.3s for this record.

I'm also going to remove the grad_scale=2 from the last record. Getting lower loss without it. (3.2765, 3.2766, 3.2776).
